### PR TITLE
BUG: .isin on datetimelike indexes do not validate input of level parameter

### LINKED ISSUE #26675: BUG: .isin on datetimelike indexes do not validate input of level parameter
#### Code Sample, a copy-pastable example if possible

```python
index = pd.DatetimeIndex([], name='foo')
index.isin([], level='foobar')

```
#### Problem description

on master:
```
array([], dtype=bool)
```

on 0.23.4

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-82f8aeda6721> in <module>
      1 index = pd.DatetimeIndex([], name='foo')
----> 2 index.isin([], level='foobar')

TypeError: isin() got an unexpected keyword argument 'level'
```
#### Expected Output
to validate input and raise same message as other non-datetimelike indexes
```
KeyError: 'Level foobar must be same as name (foo)'
```

the api for `isin` is defined in #7892
#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by simonjayhawkins**
#LondonPythonSprints

- [x] closes #26675
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/conftest.py [hunk 1]
```diff
@@ -47,3 +47,37 @@ def zero(request):
...
+
+
+def _get_subclasses(cls):
+    for subclass in cls.__subclasses__():
+        yield from _get_subclasses(subclass)
+        yield subclass
+
+
+all_indexes = [index for index in ([Index] + list(set(_get_subclasses(Index))))
+               if getattr(pd, index.__name__, None) is not None]
```
Comment (WillAyd) [inline pandas/tests/indexes/conftest.py]: What is the name check for?

**Comment by simonjayhawkins** [comment]
To exclude IntegerIndex and NumericIndex classes. But ci failed. so may
need  rethink. the idea was to create a fixture that checks all registered
indexes for API checking. May just have to list them explicitly for now

**Comment by simonjayhawkins** [comment]
@WillAyd the ci failures are now fixed. but could still parametrise `all_index_empty` directly for now and decompose later as required, if it makes the code clearer.

File: pandas/tests/indexes/conftest.py (hunk 1 above)
Comment (WillAyd) [reply pandas/tests/indexes/conftest.py]: Can just simplify to `if getattr(pd, index.__name__, False)`

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor stylistic nit but otherwise lgtm

File: pandas/tests/indexes/conftest.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/indexes/conftest.py]: missed that. thanks.

**Review by WillAyd — APPROVED** [review]

File: pandas/tests/indexes/conftest.py [hunk 2]
```diff
@@ -47,3 +49,40 @@ def zero(request):
...
+
+@pytest.fixture(params=all_indexes)
+def all_index_types(request):
+    """
+    A Fixture for all indexes types. Index and subclasses in pandas namespace.
+    """
+    return request.param
+
+
+@pytest.fixture
```
Comment (jreback) [inline pandas/tests/indexes/conftest.py]: shouldn't this just work off of the indices_list? why creating extra machinery here?

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**